### ISSUE TITLE
 ModuleNotFoundError after installing a wheel package using Poetry

### ISSUE THREAD

**Issue by davidlbit** [python-poetry/poetry#8231]
## System info
- **Poetry version**: 1.5.1
- **Python version**: 3.10.6
- **OS version and name**: PopOS 22.04
- **pyproject.toml**: [Link to pyproject.toml](https://gist.github.com/davidlbit/595b13e2ec1a649c900003edf97e3829)

## Issue

I encountered an issue while trying to install a `.whl` package using Poetry, following the instructions in the documentation (https://python-poetry.org/docs/cli/#add). When attempting to install the package directly with `poetry add <package>`, I received an error message stating the following:

> Note: This error originates from the build backend and is likely not a problem with Poetry but with scipy (1.11.1) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "scipy (==1.11.1)"'.

To work around this, I manually triggered the download for the wheel file as suggested by the error message and placed it in a folder named `wheels/`. I then installed the package using `poetry add wheels/<package_name>` (e.g., `poetry add wheels/scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`).

The installation appeared to succeed, as both the `pyproject.toml` and `poetry.lock` files reflected the package installation. However, when I tried to import the installed package, I encountered a `ModuleNotFoundError`. I have confirmed that I activated the virtual environment before attempting the import.

What am I missing?

**Comment by Secrus** [comment]
If you were to attach the traceback of the error you are getting, it would be much easier for someone to debug. Taking a quick peek at Scipy, I guess it could be a Cython3-related issue, but without any traceback, I can't say for sure.

**Comment by davidlbit** [comment]
[error_log.txt](https://github.com/python-poetry/poetry/files/12158305/error_log.txt)


That's the original traceback of the error I received.

**Comment by dimbleby** [comment]
as the error message says:

> This error originates from the build backend and is likely not a problem with Poetry but with scipy (1.11.1) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "scipy (==1.11.1)"'.

**Comment by dimbleby** [comment]
> my real issue 

so please provide diagnostics relating to your real issue!

**Comment by dimbleby** [comment]
> If there are any other diagnoistics I can run

all the obvious things , please

- your pyproject.toml (what you have currently linked is not a pyproject.toml)
- what command(s) you ran
- what the output was
- etc

currently you are hopping around between several different things, and giving only fragmentary diagnostics for any of them.  it's hard to help!  please decide what it is that you are reporting and then provide steps to reproduce that problem

perhaps a better approach would be for you to produce a dockerfile in which you reproduce your issue, then you can be sure that everyone else will be able to reproduce it too

**Comment by abn** [comment]
### 1.  PEP 517 build issues

The meson build logs in logs indicate the environment in which the installation was attempted does not contain the necessarily build time dependencies for the package. See the following snippet from the logs.

```console
  ../../meson.build:82:0: ERROR: Unknown compiler(s): [['gfortran'], ['flang'], ['nvfortran'], ['pgfortran'], ['ifort'], ['ifx'], ['g95']]
  The following exception(s) were encountered:
  Running `gfortran --version` gave "[Errno 2] No such file or directory: 'gfortran'"
  Running `gfortran -V` gave "[Errno 2] No such file or directory: 'gfortran'"
  Running `flang --version` gave "[Errno 2] No such file or directory: 'flang'"
  Running `flang -V` gave "[Errno 2] No such file or directory: 'flang'"
  Running `nvfortran --version` gave "[Errno 2] No such file or directory: 'nvfortran'"
  Running `nvfortran -V` gave "[Errno 2] No such file or directory: 'nvfortran'"
  Running `pgfortran --version` gave "[Errno 2] No such file or directory: 'pgfortran'"
  Running `pgfortran -V` gave "[Errno 2] No such file or directory: 'pgfortran'"
  Running `ifort --version` gave "[Errno 2] No such file or directory: 'ifort'"
  Running `ifort -V` gave "[Errno 2] No such file or directory: 'ifort'"
  Running `ifx --version` gave "[Errno 2] No such file or directory: 'ifx'"
  Running `ifx -V` gave "[Errno 2] No such file or directory: 'ifx'"
  Running `g95 --version` gave "[Errno 2] No such file or directory: 'g95'"
  Running `g95 -V` gave "[Errno 2] No such file or directory: 'g95'"

  A full log can be found at /tmp/tmpn9_za8qq/scipy-1.11.1/.mesonpy-b3nqoli_/build/meson-logs/meson-log.txt
```

This is meson throwing the error not poetry. And in the same environment, executing `pip wheel --use-pep517 "scipy (==1.11.1)` will most definitely raise the same errors. Similar to the following.

<details>
<summary><b>pip-wheel-console.log</b></summary>
<p>

```console
$ pip wheel scipy==1.11.1
Collecting scipy==1.11.1
  Downloading scipy-1.11.1.tar.gz (56.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.0/56.0 MB 14.5 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      + meson setup /tmp/pip-wheel-vptdftu1/scipy_b3e6aded9a3d4245ab4bb96c590b7eaf /tmp/pip-wheel-vptdftu1/scipy_b3e6aded9a3d4245ab4bb96c590b7eaf/.mesonpy-3n_ujdpr/build -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-wheel-vptdftu1/scipy_b3e6aded9a3d4245ab4bb96c590b7eaf/.mesonpy-3n_ujdpr/build/meson-python-native-file.ini
      The Meson build system
      Version: 1.3.2
      Source dir: /tmp/pip-wheel-vptdftu1/scipy_b3e6aded9a3d4245ab4bb96c590b7eaf
      Build dir: /tmp/pip-wheel-vptdftu1/scipy_b3e6aded9a3d4245ab4bb96c590b7eaf/.mesonpy-3n_ujdpr/build
      Build type: native build
      Project name: SciPy
      Project version: 1.11.1
      C compiler for the host machine: ccache cc (gcc 13.2.1 "cc (GCC) 13.2.1 20231205 (Red Hat 13.2.1-6)")
      C linker for the host machine: cc ld.bfd 2.40-14
      
      ../../meson.build:1:0: ERROR: Unknown compiler(s): [['c++'], ['g++'], ['clang++'], ['nvc++'], ['pgc++'], ['icpc'], ['icpx']]
      The following exception(s) were encountered:
      Running `c++ --version` gave "[Errno 2] No such file or directory: 'c++'"
      Running `g++ --version` gave "[Errno 2] No such file or directory: 'g++'"
      Running `clang++ --version` gave "[Errno 2] No such file or directory: 'clang++'"
      Running `nvc++ --version` gave "[Errno 2] No such file or directory: 'nvc++'"
      Running `pgc++ --version` gave "[Errno 2] No such file or directory: 'pgc++'"
      Running `icpc --version` gave "[Errno 2] No such file or directory: 'icpc'"
      Running `icpx --version` gave "[Errno 2] No such file or directory: 'icpx'"
      
      A full log can be found at /tmp/pip-wheel-vptdftu1/scipy_b3e6aded9a3d4245ab4bb96c590b7eaf/.mesonpy-3n_ujdpr/build/meson-logs/meson-log.txt
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 23.3.2 -> 24.0
[notice] To update, run: python3 -m pip install --upgrade pip
```

</p>
</details>

### 2. Wheel based installation imports

This cannot be re-produced with the latest Poetry releases. If it was a real problem, it seems it is not anymore. See example below.

```sh
podman run --rm -i --entrypoint bash docker.io/python:3.11 <<EOF
set -xe
python -m pip install --disable-pip-version-check -q poetry

poetry config virtualenvs.in-project true
poetry config virtualenvs.options.no-pip true
poetry config virtualenvs.options.no-setuptools true

poetry new demo
pushd demo
sed -i 's/python =.*/python = "~3.11"/g' pyproject.toml
pip wheel scipy==1.11.1 --no-deps
mkdir -p wheels/scipy
mv *.whl wheels/scipy/.
poetry add ./wheels/scipy/*.whl
poetry run python -c 'import scipy; print(scipy.__file__)'
EOF
```

<details>
<summary><b>console.log</b></summary>
<p>

```console
+ python -m pip install --disable-pip-version-check -q poetry
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
+ poetry config virtualenvs.in-project true
+ poetry config virtualenvs.options.no-pip true
+ poetry config virtualenvs.options.no-setuptools true
+ poetry new demo
Created package demo in demo
/demo /
+ pushd demo
+ sed -i 's/python =.*/python = "~3.11"/g' pyproject.toml
+ pip wheel scipy==1.11.1 --no-deps
Collecting scipy==1.11.1
  Downloading scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.1/59.1 kB 2.0 MB/s eta 0:00:00
Downloading scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (36.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.2/36.2 MB 20.7 MB/s eta 0:00:00
Saved ./scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+ mkdir -p wheels/scipy
+ mv scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl wheels/scipy/.
+ poetry add ./wheels/scipy/scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
Creating virtualenv demo in /demo/.venv

Updating dependencies
Resolving dependencies...

Package operations: 2 installs, 0 updates, 0 removals

  - Installing numpy (1.26.4)
  - Installing scipy (1.11.1 /demo/wheels/scipy/scipy-1.11.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl)

Writing lock file
+ poetry run python -c 'import scipy; print(scipy.__file__)'
/demo/.venv/lib/python3.11/site-packages/scipy/__init__.py
```

</p>
</details>

You will see above that the import for scipy was successful as expected.

The issue experienced by the reporter is likely related to their python environment, for example a misconfigured PYTHONPATH env var etc. It looks like for `rpds` the package might be bad or the relative import problematic.

In any case based on the above, I am closing this issue. If the issue recurs, please raise a new issue with more information and specifics around how best to reproduce it reliably.
